PostHog/PostHogFeatureFlags.swift

```diff
@@ -99,9 +99,9 @@ class PostHogFeatureFlags {
                 else {
                     hedgeLog("Error: Decide response missing correct featureFlags format")
 
-                    self.notifyAndRelease()
+                    callback([])
 
-                    return callback()
+                    return self.notifyAndRelease()
```

Comment:
the callback can be executed after releasing the locks, or is there a reason to change this?